R/RtGam.R

```diff
@@ -45,23 +55,28 @@ RtGam <- function(cases,
                   reference_date,
                   group = NULL,
                   k = smooth_dim_heuristic(length(cases)),
-                  m = penalty_dim_heuristic(length(unique(reference_date)))) {
+                  m = penalty_dim_heuristic(length(unique(reference_date))),
+                  backend = "gam",
+                  ...) {
   check_required_inputs_provided(
     cases,
     reference_date,
     group,
     k,
-    m
+    m,
+    backend
   )
   validate(cases, reference_date, group, k, m)
 
-  df <- prepare_inputs(cases, reference_date, group)
+  df <- dataset_creator(cases, reference_date, group, backend)
   formula <- formula_creator(
     k = k,
     m = m,
     is_grouped = !rlang::is_null(group)
   )
 
+  fit <- fit_model(df, formula, ...)
```

Comment:
I think i was expecting you to call `do.call(fit_model, c(list(formula), list(...)))` vs managing it internally in a method (my expectation is that this would allow for overriding defaults but perhaps that is a mistaken view?

Reply:
This worked like an absolute charm. You nailed it.